### ISSUE TITLE
[pro] Authorization issue with fallback and gate.owner

### ISSUE THREAD

**Issue by bessey** [rmosolgo/graphql-ruby#1420]
I am attempting to write a custom CanCan based auth strategy, that works similarly to Pundit in that it is capable of performing auth in the compile time "access" phase by guessing the class of each field's return type. However I am finding that I cannot do anything useful, because the only gate I ever get back is owned by the schema.

Given a custom auth strategy of:
```ruby
module Support
  class AuthStrategy
    def initialize(ctx)
    end

    def allowed?(gate, value)
      puts "ALLOWED #{gate.level} #{gate.owner} #{value}"
      true
    end
  end
end
```

And a schema auth setup of:
```ruby
  authorization(Support::AuthStrategy, fallback: {access: :read_api})
```

When I perform a query, all I see is X rows of this where X is the number of (it appears) fields in the query:

```
ALLOWED access #<GraphQL::Schema:0x00007feac6d2b870>
ALLOWED access #<GraphQL::Schema:0x00007feac6d2b870>
ALLOWED access #<GraphQL::Schema:0x00007feac6d2b870>
ALLOWED access #<GraphQL::Schema:0x00007feac6d2b870>
ALLOWED access #<GraphQL::Schema:0x00007feac6d2b870>
ALLOWED access #<GraphQL::Schema:0x00007feac6d2b870>
```

This happens to me on GraphQL Pro 1.6.5 and 1.7.4 

The root cause appears to stem from  `graphql-pro-1.7.4/lib/graphql/pro/access/define.rb#47`
Fallback is implemented to share a single gate between all fields, which of course has no knowledge of its owner.

This seems wrong to me, but I am guessing I can hack around this using an instrumenter to implement fallback instead of the auth framework.

**Comment by rmosolgo** [comment]
Thanks again for the careful writeup, @bessey! I've just released `1.7.5` which makes a new gate for each otherwise-uncovered field, so `gate.owner` will point to that field. Want to give it a try?

**Comment by bessey** [comment]
Thank you for your work! Unfortunately due to other constraints, we ended up using the Instrumenter API directly instead of `fallback`, so this is no longer an issue for me. Nevertheless that definitely sounds like the solution I needed, so feel free to close.
